### DISCUSSION TITLE
Should a disabled plugins during server runtime stop it?

### DISCUSSION THREAD

**Discussion by atyrode** [itzg/docker-minecraft-server#2420] [Ideas]
Hello,

I've realized that some issues related to plugins can happen, specifically this one:

```log
mcserv  | [02:46:31] [Server thread/INFO]: Enabling Citizens v2.0.33-SNAPSHOT (build 3216)
mcserv  | [02:46:31] [Server thread/INFO]: Loading external libraries
mcserv  | [02:46:33] [Server thread/INFO]: Creating file: config.yml
mcserv  | [02:46:33] [Server thread/INFO]: Creating file: saves.yml
mcserv  | [02:46:33] [Server thread/INFO]: Creating file: shops.yml
mcserv  | [02:46:33] [Server thread/INFO]: Enabling LibsDisguises v10.0.38
mcserv  | [02:46:33] [Server thread/INFO]: File Name: 81.jar
mcserv  | [02:46:33] [Server thread/INFO]: Discovered nms version: v1_20_R2 (v1_20_R2)
mcserv  | [02:46:33] [Server thread/INFO]: Jenkins Build: #1253
mcserv  | [02:46:33] [Server thread/INFO]: Build Date: 06/10/2023 01:50
mcserv  | [02:46:33] [Server thread/INFO]: If you own the plugin, place the premium jar downloaded from https://www.spigotmc.org/resources/libs-disguises.32453/ in plugins/LibsDisguises/
mcserv  | [02:46:33] [Server thread/INFO]: You are running the free version, commands limited to non-players and operators. (Console, Command Blocks, Admins)
mcserv  | [02:46:33] [Server thread/ERROR]: !! May I have your attention please !!
mcserv  | [02:46:33] [Server thread/ERROR]: An update for ProtocolLib has been downloaded and will be installed when the server restarts. When possible, please restart the server. Lib's Disguises may not work correctly until you do so.
mcserv  | [02:46:33] [Server thread/ERROR]: !! May I have your attention please !!
mcserv  | [02:46:33] [Server thread/ERROR]: Error occurred while enabling LibsDisguises v10.0.38 (Is it up to date?)
(... extensive stacktrace ...)
mcserv  | [02:46:33] [Server thread/INFO]: Disabling LibsDisguises v10.0.38
```

When a plugin failed to load at server runtime, it becomes non deterministic, because the image runs fine, but something *failed* unexpectedly and wasn't caught by the expected CI of running this image.

Many things can fail, but a plugin is defined before server runtime, and the auto-update that SPIGET_RESOURCES has can be a source of unexpected issues and behaviors, I'm thinking that there should be an ENV variables that defines whether or not the image should accept plugins failing to load. And if we specifically said we don't accept it, it kills the container.

This could be detected through reading the stdout of the server process; or by using the rcon client and reading the /plugins output I guess?

I'm not opening an issue/enhancement just yet because I'm not convinced on wether the paradigm should be to be robust or error-free, but I guess having the ability to define it wouldn't hurt anyone.

**Comment by itzg** [comment]
That seems like a very difficult concept to implement. It also goes against a core principle of this image where all it can control is the setup of the working directory. After that it literally exec's off to the Java Minecraft process and gets out of the way. (Auto-pause is only slight exception to that paradigm.)

Now if this is something that could have been observed from the spigot metadata, then that's something that could be in scope for the image setup.
